datastructures/clockchain.py

```diff
@@ -184,20 +270,58 @@ def latest_selected_tick(self):
 
         return tick
 
+    def purge_by(self, candidates, func):
+        max_val = func(
+            max(
+                candidates,
+                key=func
+            )
+        )
+        return [
+            candidate
+            for candidate in candidates
+            if func(candidate) == max_val
+        ]
+
+    def num_pings(self, tick):
+        return len(tick['list'])
+
+    def hash_diff(self, tick):
+        pkhash = hasher({'0': tick['pubkey']})
```

Comment:
we need to standardize this somehow, too hacky to do a dict on '0'

Reply:
solution: Make Hasher understand how to use non-dicts as well (not just rely on everything being a dict..)

Reply:
Added a todo to this effect in the codebase